deploy/examples/csi/rbd/storageclass-radosnamespace.yaml

```diff
@@ -0,0 +1,65 @@
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: replicapool
+  namespace: rook-ceph # namespace:cluster
+spec:
+  failureDomain: osd
+  replicated:
+    size: 3
+    # Disallow setting pool with replica 1, this could lead to data loss without recovery.
+    # Make sure you're *ABSOLUTELY CERTAIN* that is what you want
+    requireSafeReplicaSize: true
+    # gives a hint (%) to Ceph in terms of expected consumption of the total cluster capacity of a given pool
+    # for more info: https://docs.ceph.com/docs/master/rados/operations/placement-groups/#specifying-expected-pool-size
+    #targetSizeRatio: .5
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPoolRadosNamespace
+metadata:
+  name: namespace-a
+  namespace: rook-ceph # namespace:cluster
+spec:
+  # The name of the CephBlockPool CR where the namespace is created.
+  blockPoolName: replicapool
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: rook-ceph-block
+provisioner: rook-ceph.rbd.csi.ceph.com # csi-provisioner-name
+parameters:
+  # normally, clusterID is the kubernetes namespace where the rook cluster is running
+  # But putting the ClusterID of  a CephBlockPoolRadosNamespace CR here instead
+  # will let the storageClass consume that Rados namespace.
+  clusterID: 80fc4f4bacc064be641633e6ed25ba7e
```

Comment:
will this be always same for this example setup, if not we should not hardcode this

Reply:
best I know yes, it will be the same (being a hash of the blockpool name etc). this is why I put it into the example file.

Reply:
Okay that's great, seems constant https://github.com/rook/rook/blob/517b1bf1283e0698cd2a0417c5023ae393e491b2/pkg/operator/ceph/pool/radosnamespace/controller.go#L362
I'm not sure why we decided to have a hash, instead we can always use the full name. Or request a new field from csi for enrolling radosnamesapce

Reply:
@parth-gr the line you are copying is missing the surrounding context.

https://github.com/rook/rook/blob/517b1bf1283e0698cd2a0417c5023ae393e491b2/pkg/operator/ceph/pool/radosnamespace/controller.go#L361-L364

Notice that the function here (`buildClusterID()`) returns `Hash(clusterID)`. Thus, the ID **is** the hashed form of the name.